Sources/NIOPosix/System.swift

```diff
@@ -756,7 +759,21 @@ internal enum Posix {
         }
     }
 
+    @inline(never)
+    public static func mkdir(pathname: String, mode: mode_t) throws {
+        _ = try syscall(blocking: false) {
+            sysMkdir(pathname, mode)
+        }
+    }
+
 #if os(macOS) || os(iOS) || os(watchOS) || os(tvOS)
+    @inline(never)
+    public static func mkpath(pathname: String, mode: mode_t) throws {
```

Comment:
This function should be called `mkpath_np`.

Reply:
fixed